cnd/src/http_api/routes.rs

```diff
@@ -148,6 +291,68 @@ pub struct BobHanEthereumHalightBitcoinState {
     pub finalized_swap: comit_ln::FinalizedSwap,
 }
 
+impl Status for BobHanEthereumHalightBitcoinState {
+    fn status(&self) -> SwapStatus {
+        // TODO: see `http_api/routes/rfc003/swap_state.rs` `SwapStatus::new`
+        unimplemented!()
+    }
+}
+
+impl MyRole for BobHanEthereumHalightBitcoinState {
+    fn my_role(&self) -> Role {
+        Role::Bob
+    }
+}
+
+impl QuantityWei for BobHanEthereumHalightBitcoinState {
+    fn quantity_wei(&self) -> String {
+        self.finalized_swap.alpha_asset.to_wei_dec()
+    }
+}
+
+impl QuantitySatoshi for BobHanEthereumHalightBitcoinState {
+    fn quantity_satoshi(&self) -> String {
+        self.finalized_swap.beta_asset.as_sat().to_string()
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct HanEthereum {
+    pub protocol: String,
+    pub quantity: String, // In Wei.
+}
+
+#[derive(Debug, Serialize)]
+struct HalightBitcoin {
+    pub protocol: String,
+    pub quantity: String, // In Satoshi.
+}
+
+#[derive(Debug, Serialize)]
+struct Transaction {
+    transactions: Vec<StepTransactions>,
+    status: EscrowStatus,
+}
+
+/// Key value store where values are transaction ids and keys are one of:
+/// "init", "deploy", "fund", "redeem", "refund"
```

Comment:
Make it an enum then? :)

Reply:
I was working under the assumption that changes to the API were out of scope for this ticket.  Is this assumption correct?  If not there are probably a few minor tweaks we can do to the schema defined in the js-sdk.  Would it be useful to write a `schema.json` file instead of coding to the js-sdk (or was that decided against already?)